src/main/scala/com/reagroup/appliedscala/urls/savemovie/SaveMovieController.scala

```diff
@@ -18,7 +17,17 @@ class SaveMovieController(saveNewMovie: NewMovieRequest => IO[ValidatedNel[Movie
     * 3. Pattern match and convert every case into an HTTP response. To Pattern match on `Validated`, use `Invalid` and `Valid`.
     * Hint: Use `Created(...)` to return a 201 response when the movie is successfully saved and `BadRequest(...)` to return a 403 response when there are errors.
     */
-  def save(req: Request[IO]): IO[Response[IO]] =
-    ???
-
+  def save(req: Request[IO]): IO[Response[IO]] = {
```

Comment:
Answer key did it like this:
```scala
  def save(req: Request[IO]): IO[Response[IO]] = for {
    newMovieRequest <- req.as[NewMovieRequest]
    maybeMovie <- saveNewMovie(newMovieRequest).attempt
    response <- maybeMovie match {
      case Left(error) => ErrorHandler(error)
      case Right(Valid(movieId)) => Created(movieId)
      case Right(Invalid(validationErrors)) => BadRequest(validationErrors.asJson)
    }
  } yield response
```

Reply:
I was surprised that you could do `req.as[NewMovieRequest]`  as `req` was not JSON type